doc/architecture/decisions/0004-upgrade-testing.md

```diff
@@ -0,0 +1,39 @@
+# 3. upgrade testing methodology when releasing
+
+Date: 2020-06-25
+
+## Status
+
+Accepted
+
+## Context
+
+One of the goals of the KubeCF project is to offer a smooth experience when it comes to upgrading Cloud Foundry deployments.
+That means any user operating a KubeCF deployment should:
+
+- be able to easily identify which versions he can upgrade to
+- upgrade using documented procedures
+- have confidence that the exact version upgrade he's doing has been tested by the KubeCF team 
+
+## Decision
+
+- not allowed to have minor releases for version `X.y.z` if `X+1` has been released (only patch releases)
+- for a version `X.y.z`, the upgrade path must be tested for:
+  - the previous `X.*.*` release
+  - version max(`X-1.*.*`)
+
+For downstream projects, assuming they don't release as often as KubeCF, the following rules should be in place:
```

Comment:
Nitpicking:
```suggestion
Therefore, for downstream projects, assuming they don't release as often as KubeCF, we recommend that the following rules should be in place:
```

As downstream actors are free to consume KubeCF as they see fit.